tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,125 @@
+import unittest
+from unittest import mock
+from unittest.case import TestCase
+from requests.exceptions import Timeout
+import tap_quickbooks
+from tap_quickbooks import LOGGER, QuickbooksClient
+
+class TestBackoffError(unittest.TestCase):
+    '''
+    Test that backoff logic works properly.
+    '''
+    @mock.patch('tap_quickbooks.client.OAuth2Session.request')
+    @mock.patch('tap_quickbooks.client.QuickbooksClient.get')
+    def test_request_timeout_and_backoff_on_make_request(self, mock_get, mock_send):
+        """
+        Check whether the request backoffs properly for 5 times in case of Timeout error.
+        """
+        mock_send.side_effect = Timeout
+        with self.assertRaises(Timeout):
+            client = QuickbooksClient("", {"start_date": "dummy_start_date", "refresh_token": "dummy_token", "client_id": "dummy_client_id", "client_secret": "dummy_client_secret", "user_agent": "dummy_ua", "realm_id": "dummy_ri"})
```

Comment:
I think this line should be moved before the `with` statement so it's clear what method should be timing out.

Reply:
updated